pkg/virtual/framework/dynamic/types.go

```diff
@@ -31,32 +29,13 @@ var _ framework.VirtualWorkspace = (*DynamicVirtualWorkspace)(nil)
 // DynamicVirtualWorkspace is an implementation of a framework.VirtualWorkspace which can dynamically serve resources,
 // based on API definitions (including an OpenAPI v3 schema), and a Rest storage provider.
 type DynamicVirtualWorkspace struct {
-	Name             string
-	RootPathResolver framework.RootPathResolverFunc
-	Authorizer       authorizer.AuthorizerFunc
-	Ready            framework.ReadyFunc
+	framework.Namer
+	framework.RootPathResolver
```

Comment:
Not to throw so bikeshedding here, but RootPathResolver does not IMO encapsulate all the requirements of this function. 

This is required to also determine the APIDomainKey and set it for dynamic but the FixedGroupVersions does not need this. 

I would be wondering if two functions might be better here?

Reply:
Valid concern but it seems orthogonal to this PR.